tests/provision/container/toolbox/test.sh

```diff
@@ -0,0 +1,159 @@
+#
+
+rlJournalStart
+    rlPhaseStartSetup
+        rlRun "toolbox_container_name=\$(uuidgen)" 0 "Generate toolbox container name"
+
+        # When running via tmt, we will use a new user to run toolbox
+        if env | grep -Eq "(PACKIT_COPR_PROJECT|TMT_TREE)"; then
+          rlRun "toolbox_user=toolbox" 0 "Generate toolbox container name"
+        fi
+    rlPhaseEnd
+
+    rlPhaseStartTest "Create toolbox container"
+        # When run via tmt, we will use a test user, toolbox under root does not work well
+        if env | grep -Eq "(PACKIT_COPR_PROJECT|TMT_TREE)"; then
+
+          # Add a toolbox user
+          rlRun "useradd $toolbox_user"
+          rlRun "toolbox_user_id=$(id -u $toolbox_user)"
+
+          # Make sure systemd user session runs for the new user
+          rlRun "loginctl enable-linger $toolbox_user"
+
+          # Add required environment variables to the user
+          rlRun "echo export XDG_RUNTIME_DIR=/run/user/$toolbox_user_id >> /home/$toolbox_user/.bashrc"
+          rlRun "echo export DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/$toolbox_user_id/bus >> /home/$toolbox_user/.bashrc"
+
+          rlRun "sudo -iu $toolbox_user toolbox create -y $toolbox_container_name"
+
+        # When running manually ./test.sh, just create the toolbox, no additional setup needed
+        else
+          rlRun "toolbox create -y $toolbox_container_name"
+        fi
+    rlPhaseEnd
+
+    toolbox_run() {
+      if [ -n "$toolbox_user" ]; then
+         local command="sudo -iu $toolbox_user toolbox run --container $toolbox_container_name $*"
+      else
+         local command="toolbox run --container $toolbox_container_name $*"
+      fi
+      echo "Command: $command"
+      eval "$command"
+    }
+
+    # Execution in Packit via tmt
+    # https://packit.dev/docs/configuration/upstream/tests#environment-variables
+    if env | grep -q PACKIT_COPR_PROJECT; then
+        rlPhaseStartTest "Packit execution: Install tmt in from copr repository"
+            TMT_COMMAND=tmt
+
+            rlRun "type toolbox_run"
+
+            # Install tmt from the copr repository, we need only the container provisioner
+            rlRun "toolbox_run sudo dnf -y install dnf-plugins-core"
+            rlRun "toolbox_run sudo dnf -y copr enable $PACKIT_COPR_PROJECT"
+            rlRun "toolbox_run sudo dnf -y install tmt+provision-container"
+        rlPhaseEnd
+
+    # Execution locally via tmt
+    elif env | grep -q TMT_TREE; then
+        rlPhaseStartTest "Local execution via tmt: Install tmt from TMT_TREE"
+            TOOLBOX_TREE="/var/tmp/tree"
+            TMT_COMMAND="env -C ${TOOLBOX_TREE} hatch -e dev run env -C /tmp tmt"
+
+            rlRun "type toolbox_run"
+
+            # Install all dependencies needed for hatch installation
+            rlRun "toolbox_run sudo dnf -y install git gcc hatch krb5-devel libpq-devel libvirt-devel python-devel"
```

Comment:
What about using `make develop` instead of duplicating the list here?

Reply:
@psss yeah, I can try, is a problem to add to `make develop` also `hatch`?

Reply:
`make develop` calls `make _deps` which installs `hatch python3-devel python3-hatch-vcs rpm-build`.

Reply:
@psss used `make develop`